packages/driver/src/multi-domain/communicator.ts

```diff
@@ -192,14 +196,14 @@ export class SpecBridgeDomainCommunicator extends EventEmitter {
    * @param {any} data - any meta data to be sent with the event.
    */
   toPrimary (event: string, data?: any, options = { syncConfig: false }) {
-    debug('<= to Primary ', event, data, location.hostname)
+    debug('<= to Primary ', event, data, window.specBridgeDomain)
     if (options.syncConfig) this.syncConfigEnvToPrimary()
 
     this.handleSubjectAndErr(data, (data: any) => {
       this.windowReference.top.postMessage({
         event: `${CROSS_DOMAIN_PREFIX}${event}`,
         data,
-        domain: location.hostname,
```

Comment:
The host name is converted into punycode, which will not match the domain specified in the command. To circumvent this and avoid adding a dependency to convert punycode to unicode I have attached the unicode domain name to the window via the server.